src/lib/grad.jl

```diff
@@ -14,3 +14,24 @@ macro which(ex)
   @capture(ex, f_(args__)) || error("Zygote.@which f(args...)")
   :(InteractiveUtils.@which adjoint(Context(), $(esc(f)), $(esc.(args)...)))
 end
+
+"""
+
+    checkpointed(f, xs...)
+
+Use gradient checkpointing on the call `f(xs...)`. This means that
+`checkpointed(f, xs...) === f(xs...)`, but when computing the derivative
+intermediate results from the forward pass of `f` will not be stored. Instead the forward
+pass will be repeated, when computing the derivative.
+This this saves memory at the cost of inreasing exectution time.
+"""
+checkpointed(f, xs...) = f(xs...)
+
+function Zygote._pullback(ctx::Zygote.AContext, ::typeof(checkpointed), f, xs...)
```

Comment:
Implementation taken from @willtebbutt in #884

Reply:
Is `lib/grad.jl`the file where this should live? Should I export?

Reply:
@DhairyaLGandhi would you want to export this functionality?

Reply:
Hmm, I think it's a bit generic sounding to export. It's not obvious which context it is expected to run in.